modin/test/storage_formats/pandas/test_internals.py

```diff
@@ -757,3 +757,81 @@ def test_repartitioning(set_num_partitions):
+    ----------
+    col_name : {"numeric_col", "non_numeric_col"}
+        The tested function takes a key column name to which the pivot values belong.
+        The function may behave differently depending on the type of that column.
+    ascending : {True, False}
+        The split parts are returned either in ascending or descending order.
+        This parameter helps us to test both of the cases.
+    num_pivots : {3, 2, 1}
+        The function's behavior may depend on the number of boundary values being passed.
+    all_pivots_are_unique : {True, False}
+        Duplicate pivot values cause empty partitions to be produced. This parameter helps
+        to verify that the function still behaves correctly in such cases.
+    """
+    from modin.core.dataframe.pandas.dataframe.utils import (
+        split_partitions_using_pivots_for_sort,
+    )
+
+    random_state = np.random.RandomState(42)
+
+    df = pandas.DataFrame(
+        {
+            "numeric_col": range(9),
+            "non_numeric_col": list("abcdefghi"),
+        }
+    )
+    min_val, max_val = df[col_name].iloc[0], df[col_name].iloc[-1]
+
+    # Selecting random boundary values for the key column
+    pivots = random_state.choice(df[col_name], num_pivots, replace=False)
+    if not all_pivots_are_unique:
+        # Making the 'pivots' contain only duplicate values
+        pivots = np.repeat(pivots[0], num_pivots)
+    # The tested function assumes that we pass pivots in the ascending order
+    pivots = np.sort(pivots)
+
+    # Randomly reordering rows in the dataframe
+    df = df.reindex(random_state.permutation(df.index))
+    bins = split_partitions_using_pivots_for_sort(df, df, col_name, pivots, ascending)
+
+    # Building reference bounds to make the result verification simpler
+    bounds = np.concatenate([[min_val], pivots, [max_val]])
+    if not ascending:
+        # If the order is descending we want bounds to be in the descending order as well:
+        # Ex: bounds = [0, 2, 5, 10] for ascending and [10, 5, 2, 0] for descending.
+        bounds = bounds[::-1]
+
+    for idx, part in enumerate(bins):
+        if ascending:
+            # Check that each part is in the range of 'bound[i] <= part <= bound[i + 1]'
+            # Example, if the `pivots` were [2, 5] and the min/max values for the colum are min=0, max=10
+            # Then each part satisfies: 0 <= part[0] <= 2; 2 <= part[1] <= 5; 5 <= part[2] <= 10
+            assert (
+                (bounds[idx] <= part[col_name]) & (part[col_name] <= bounds[idx + 1])
+            ).all()
+        else:
+            # Check that each part is in the range of 'bound[i + 1] <= part <= bound[i]'
+            # Example, if the `pivots` were [2, 5] and the min/max values for the colum are min=0, max=10
+            # Then each part satisfies: 5 <= part[0] <= 10; 2 <= part[1] <= 5; 0 <= part[2] <= 2
+            assert (
+                (part[col_name] <= bounds[idx]) & (part[col_name] >= bounds[idx + 1])
```

Comment:
This makes it easier to read and matches the comment above.
```suggestion
                (bounds[idx + 1] <= part[col_name) & (part[col_name] <= bounds[idx])
```